sunpy/database/tests/test_database.py

```diff
@@ -441,6 +441,26 @@ def test_add_entry_from_hek_qr(database):
     assert len(database) > 1
 
 
+@pytest.mark.remote_data
+def test_hek_query_download(database, tmpdir):
+
+    assert len(database) == 0
+
+    query = hek.HEKClient().search(
+        hek.attrs.Time('2019/03/10 14:40:10', '2019/04/11 16:40:50'),
+        hek.attrs.EventType('FL')
+    )
+
+    database.download_from_hek_query_result(
+        query[4], path=str(tmpdir.join('{file}.fits')))
+
+    fits_file_pattern = str(tmpdir.join('*.fits'))
+    num_of_fits_headers = sum(
+        len(fits.get_header(file)) for file in glob.glob(fits_file_pattern))
+
+    assert len(database) > 0 and len(database) == num_of_fits_headers
```

Comment:
How many files does this download and how large?

Reply:
10 files with total of around 100 megabytes

Reply:
There is no way to make this smaller?

Reply:
I tried many queries but couldn't find something smaller

Reply:
One more method would be to keep the files in sample and mock parfive.download to return path to those files instead.

Reply:
I don't think adding these files to sample data is the way forward here. I think somehow mocking the download and getting fake files would be better but I am unsure how to mock the download.

Reply:
We need headers from the files since the database entries are made from them.

Reply:
You mean fits headers? We can probably mock those too.

Reply:
One more way would be to mock the function ``` entries_from_file ``` to return mock DatabaseEntry objects [here](https://github.com/sunpy/sunpy/blob/master/sunpy/database/database.py#L419).

Reply:
We also don't want to download any files if we can help it or at least not that many.

Reply:
The above method won't download any file. But we can't really test then if the files are downloaded or not.

Reply:
Whatever works to lessen downloading 10 files.

Reply:
So should I do the ```entries_from_file``` mocking thing ?

Reply:
If that will work.